includes/base/widget-base.php

```diff
@@ -204,10 +355,24 @@ public function render_content() {
 		<?php
 	}
 
+	/**
+	 * Render widget plain content.
+	 *
+	 * Output the widget final HTML on the frontend.
```

Comment:
Used to save generated HTML to the DB, in order to allow search, and access to the content even if Elementor is deactivated.  if the widget doesn't display any content , like form widgets, return an empty string because there is no content to return and if Elementor is deactivated there won't be any form to display. If the widget uses shortcodes to display the data, The best practis is to return the shortcode itself.